src/datasets/arrow_dataset.py

```diff
@@ -3577,6 +3619,7 @@ def push_to_hub(
         token: Optional[str] = None,
         branch: Optional[str] = None,
         shard_size: Optional[int] = 500 << 20,
+        allow_cast: bool = True,
```

Comment:
We can name this parameter embed_external_files to keep things simple if you want